src/dosext/mfs/mfs.c

```diff
@@ -716,27 +716,73 @@ int dos_utime(char *fpath, struct utimbuf *ut)
   return -1;
 }
 
-int dos_get_disk_space(const char *cwd, unsigned int *free, unsigned int *total,
-		       unsigned int *spc, unsigned int *bps)
+static int dos_get_disk_space(const char *cwd, unsigned int *dfree,
+                              unsigned int *total, unsigned int *spc,
+                              unsigned int *bps)
 {
   struct statfs fsbuf;
 
-  if (statfs(cwd, &fsbuf) >= 0) {
-    /* return unit = 512-byte blocks @ 1 spc, std for floppy */
-    *spc = 1;
-    *bps = 512;
-    *free = (fsbuf.f_bsize / 512) * fsbuf.f_bavail;
-    *total = (fsbuf.f_bsize / 512) * fsbuf.f_blocks;
-
-    while (*spc < 64 && *total > 65535) {
-      *spc *= 2;
-      *free /= 2;
-      *total  /= 2;
-    }
-    return (1);
-  }
-  else
-    return (0);
+  if (statfs(cwd, &fsbuf) < 0)
+    return 0;
+
+  /* return unit = 512-byte blocks @ 1 spc, std for floppy */
+  *spc = 1;
+  *bps = 512;
+  *dfree = (fsbuf.f_bsize / 512) * fsbuf.f_bavail;
+  *total = (fsbuf.f_bsize / 512) * fsbuf.f_blocks;
```

Comment:
Please use (bps * spc) here instead of hardcoding 512.
Also you need to do multiplication first, divizion second
to avoid rounding errors.

Reply:
I only moved that code from lfn.c.

Reply:
Yes, I see, but rounding errors can represent
a severe problem, so why not to fix while you
are at it?